synapse/handlers/search.py

```diff
@@ -63,11 +63,15 @@ def get_old_rooms_from_upgraded_room(self, room_id):
         while True:
             try:
                 predecessor = yield self.store.get_room_predecessor(room_id)
-                predecessor_room_id = predecessor["room_id"]
+                if not predecessor:
+                    # This room does not have a predecessor
+                    break
             except NotFoundError:
-                # This room does not have a predecessor
+                # Unknown room_id
                 break
 
+            predecessor_room_id = predecessor["room_id"]
```

Comment:
given that `predecessor` comes from an event, how do you know it is a dict with a `room_id` member?

Reply:
There are some other poor uses of this method about. I'll add some more error handling to `get_room_predecessor`.

https://github.com/matrix-org/synapse/blob/2173785f0d9124037ca841b568349ad0424b39cd/synapse/handlers/federation.py#L1187-L1192